bpython/autocomplete.py

```diff
@@ -335,6 +335,48 @@ def list_attributes(self, obj):
                 return dir(obj)
 
 
+class ArrayItemMembersCompletion(BaseCompletionType):
+
+    def __init__(self, shown_before_tab=True, mode=SIMPLE):
+        self._shown_before_tab = shown_before_tab
+        self.completers = CumulativeCompleter((
+                            StringLiteralAttrCompletion(mode=mode),
+                            AttrCompletion(mode=mode),
+                            ParameterNameCompletion(mode=mode)),
+                            mode=mode)
+
+    def matches(self, cursor_offset, line, **kwargs):
+        if 'locals_' not in kwargs:
+            return None
+        locals_ = kwargs['locals_']
+
+        r = self.locate(cursor_offset, line)
+        if r is None:
+            return None
+        member_part = r[2]
+        _, _, dexpr = lineparts.current_array_with_indexer(cursor_offset, line)
+        try:
+            exec('temp_val_from_array = ' + dexpr, locals_)
```

Comment:
I wasn't sure how to do it with safe_eval. From my understanding, safe_eval doesn't allow me to set a temp variable dynamically, it will just return a real variable value. I needed a variable name in locals_ that I could reference by string in completers like AttrCompletion, so I could make use of their logic without any changes to their code.